analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -241,10 +241,19 @@ private void VerifyDispose(MethodDeclarationSyntax disposeMethod, bool isSealedC
                 (semanticModel.GetSymbolInfo(baseType.Type).Symbol as INamedTypeSymbol).Is(KnownType.System_IDisposable);
 
             private bool CallsSuppressFinalize(BaseMethodDeclarationSyntax methodDeclaration) =>
-                methodDeclaration.ContainsMethodInvocation(semanticModel, method => HasArgumentValues(method, "this"), KnownMethods.IsGcSuppressFinalize);
-
-            private bool CallsVirtualDispose(BaseMethodDeclarationSyntax methodDeclaration, string argumentValue) =>
-                methodDeclaration.ContainsMethodInvocation(semanticModel, method => HasArgumentValues(method, argumentValue), IsDisposeBool);
+                methodDeclaration.ContainsMethodInvocation(semanticModel,
+                    method => method.Expression.GetName() == nameof(GC.SuppressFinalize)
```

Comment:
Instead of using `.GetName()` and equality check, we can use `.NameIs` extension method